cohortextractor/main.py

```diff
@@ -1,31 +1,25 @@
 import csv
-import importlib
+import importlib.util
 import inspect
-import os
-import sys
-from pathlib import Path
 
 from .backends import BACKENDS
 from .query_utils import get_column_definitions
 
 
-def main(workspace="/workspace", backend_id=None, db_url=None):
-    if not db_url:
-        db_url = os.environ["TPP_DATABASE_URL"]
-    if not backend_id:
-        backend_id = os.environ["BACKEND"]
+def main(workspace, definition, backend_id, db_url):
```

Comment:
Would names like `workspace_path` and `definition_path` make it clearer what this function accepts?

Reply:
Are you including "path" there to indicate the _type_ of the thing (a `Path`), or because it's the "path of" the workspace and definition that's being passed in? I could get behind the latter (especially for the definition), but if we feel we need type annotations then I think we should be using the real thing rather than mangling the variable names.

Reply:
How would you feel about `workspace_dir` and `definition_path`?